packages/express-session/src/index.ts

```diff
@@ -0,0 +1,98 @@
+import { AccountsServer } from '@accounts/server';
+import { Tokens, User } from '@accounts/types';
+import { Request, Response, NextFunction } from 'express';
+import * as requestIp from 'request-ip';
+import 'express-session';
+import { getUserAgent } from './utils/get-user-agent';
+
+export interface AccountsSessionOptions {
+  user: {
+    name: string;
+    resolve: (tokens: Tokens) => User | Promise<User>;
+  };
+  name?: string;
+}
+
+export class AccountsSession {
+  constructor(private accountsServer: AccountsServer, private options: AccountsSessionOptions) {
+    if (!this.options.name) {
+      this.options.name = 'accounts-js-tokens';
+    }
+  }
+
+  public middleware() {
+    return async (req: Request, res: Response, next: NextFunction) => {
+      try {
+        await this.renew(req);
+
+        const tokens = this.get(req);
+        const user = await this.options.user.resolve(tokens);
+
+        req[this.options.user.name] = user;
+
+        next();
+      } catch (e) {
+        next(e);
+      }
+    };
+  }
+
+  public destroy(req?: Request, callback?: (err: any) => void): void {
+    req.session.destroy(async err => {
+      const tokens = this.get(req);
+
+      if (tokens && tokens.accessToken) {
+        await this.accountsServer.logout(tokens.accessToken);
+        await this.clear(req);
+      }
+
+      if (callback) {
```

Comment:
wouldn't be better to return a promise instead of callback? we never use callbacks in our codebase right now